objc/objc_runtime_darwin.go

```diff
@@ -98,10 +103,315 @@ func GetClass(name string) Class {
 }
 
 // AllocateClassPair creates a new class and metaclass. Then returns the new class, or Nil if the class could not be created
+// Deprecated: use RegisterClass instead
 func AllocateClassPair(super Class, name string, extraBytes uintptr) Class {
 	return objc_allocateClassPair(super, name, extraBytes)
 }
 
+// Selector is an interface that takes a Go method name
+// and returns the selector equivalent name.
+// If it returns a nil SEL then that method
+// is not added to the class object.
+type Selector interface {
+	Selector(string) SEL
+}
+
+// TagFormatError occurs when the parser fails to parse the objc tag in a Selector object
+var TagFormatError = errors.New(`objc tag doesn't match "ClassName : SuperClassName <Protocol, ...>""`)
+
+// MismatchError occurs when the Go struct definition doesn't match that of Objective-C
+var MismatchError = errors.New("go struct doesn't match objective-c struct")
+
+// RegisterClass takes a pointer to a struct that implements the Selector interface.
+// It will register the structs fields and pointer receiver methods in the Objective-C
+// runtime using the SEL returned from Selector. Any errors that occur trying to add
+// a Method or Ivar is returned as an error. Such errors may occur in parsing or because
+// the size of the struct does not match the size in Objective-C. If no errors occur
+// then the returned Class has been registered successfully.
+//
+// The struct's first field must be of type Class and have a tag that matches the format
+// `objc:"ClassName : SuperClassName <Protocol, ...>`. This tag is equal to how the class
```

Comment:
Is a space allowed between `ClassName` and `SuperClassName`? (I think that's fine)

Reply:
Yes whitespace is allowed in between. That is how it is for Objective-C class definitions